lib/portage/tests/ebuild/test_fetch.py

```diff
@@ -36,6 +38,85 @@
 
 
 class EbuildFetchTestCase(TestCase):
+
+    async def _test_interrupt(self, loop, server, async_fetch, pkg, ebuild_path):
```

Comment:
Thanks for the test case. This is all so hairy that it's good to have as much as we can.